pxteditor/code-validation/validateBlockFieldValueExists.ts

```diff
@@ -15,8 +15,13 @@ export function validateBlockFieldValueExists({ usedBlocks, fieldType, fieldValu
         block.isEnabled() && block.type === specifiedBlock
     );
 
-    const successfulBlocks = enabledSpecifiedBlocks.filter((block) =>
-        block.getFieldValue(fieldType) === fieldValue
+    const successfulBlocks = enabledSpecifiedBlocks.filter((block) => {
+        if (fieldType === "VAR") {
```

Comment:
Is this reacting to a special case in blockly, or is it a special case we're creating ourselves? I'd love a comment here explaining what's up.

Reply:
I'll make a comment. It's reacting to a special case in blockly. Performing `block.getFieldValue(fieldType)` for a variable field block will return the `id` of that variable rather than it's name, so the comparison with `fieldValue` will never be truthy. For example, in the case that we're checking that the variable being used is named `row`, the comparison for the return would be `row === -2lX#Jei3u)pzbSEnrMB`.

Reply:
Is that just a signal that we should create a separate validator for variable names rather than using field value and type?

Reply:
Yeah, I thought about that, too. I'm on the fence about it. Yes, the VAR field type uses a completely different function to reach the same thing as a different field value, _**but**_ the inputs to that validator will be the exact same. Or, I guess since we'd split it out, the variable validator wouldn't need the `fieldType` and `fieldValue` could be renamed to `varName` or something like that. 

I think it would be extremely reasonable to split it out, but being that we're still unsure of the value of validating variable names, I'm leaning towards leaving it for now. If it's something that teachers say they want, I think we can change it then.